astropy/units/quantity.py

```diff
@@ -1452,35 +1467,64 @@ def to_string(self, unit=None, precision=None, format=None, subfmt=None):
         }
         formats["latex_inline"] = formats["latex"]
 
-        if format not in formats:
```

Comment:
Maybe just leave this test in place (i.e., just remove the `elif` clause. It is not like it will matter for speed...